polyfactory/utils/predicates.py

```diff
@@ -79,7 +79,7 @@ def is_union(annotation: Any) -> "TypeGuard[Any | Any]":
 
 
 def is_optional_union(annotation: Any) -> "TypeGuard[Any | None]":
-    """Determine whether a given annotation is 'typing.Optional'.
+    """Determine whether a given annotation is 'typing.Union[XXXX, NoneType]'.
```

Comment:
Feel free to disagree, but I would rename this function to `is_optional`, I got too confused with the `_union` part. I modified the docstring to reflect that `Optional[XXX]` is the same as `Union[XXX, NoneType]`

Reply:
You can rename this